### PR TITLE
Add types of es/ListBox at carbon-components-react package

### PR DESCRIPTION
**Pull request by mango906**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

Hi! 
I have issue like this https://github.com/carbon-design-system/carbon/issues/11058
In summary, there is a problem that tree-shaking does not occur when the ListBox component of the carbon-components-react package is imported with commonJs.
So I'm trying to import a file of the components of the `es/*` folder for tree-shaking, but there are no definition files with `es/*` components. So I added definition file of ListBox components.
The only declaration file I currently need is ListBox, but should I add other files? If, necessary i will add other components though

Thanks

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## carbon-components-react ([<kbd>unpkg</kbd>](https://unpkg.com/browse/carbon-components-react@latest/))
was missing the following properties:
1. ActionableNotification
1. unstable_ContainedTab
1. unstable_ContainedTabs
1. unstable_usePrefix
1. unstable_Popover

<details>
<summary>as well as these 6 other properties...</summary>
<p>unstable_PopoverContent, unstable_HStack, unstable_Stack, unstable_VStack, unstable_Tooltip, unstable_IconButton</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d2220c18ed9e5fc361fde350d74ec80cea03c31c
</p>

**Comment by mango906** [comment]
^ I think it's related with package's migration

**Comment by andrewbranch** [comment]
Going to need owner input on this one: pinging @kalbert312 @sgregoire @matchilling again, thank you!

**Comment by jakebailey** [comment]
To be clear, these files are just copy/pastes of the `lib` files but at a new path? That seems awkward; I feel like there's a better way (and I know other DT packages do it differently, like `ramda` or `carbon__icons-react` or other things that mention `es`.

**Comment by mango906** [comment]
> To be clear, these files are just copy/pastes of the `lib` files but at a new path? That seems awkward; I feel like there's a better way (and I know other DT packages do it differently, like `ramda` or `carbon__icons-react` or other things that mention `es`.

Yes. other components are no problems to use, but ListBox Components by lib has [tree-shaking issue](https://github.com/carbon-design-system/carbon/issues/11058). But, es ListBox components doesn't support typescript definition. 
So in my project, i'm using like this.

```typescript
declare module 'carbon-components-react/es/components/ListBox' {
  import {
    ReactDivAttr,
    ForwardRefReturn,
  } from 'carbon-components-react/typings/shared';

  import { ListBoxFieldComponent } from './ListBox/ListBoxField';
  import { ListBoxMenuComponent } from './ListBox/ListBoxMenu';
  import { ListBoxMenuIconComponent } from './ListBox/ListBoxMenuIcon';
  import { ListBoxMenuItemComponent } from './ListBox/ListBoxMenuItem';

  type ExcludedAttributes = 'onKeyDown' | 'onKeyPress' | 'ref';

  export interface ListBoxProps extends Omit<ReactDivAttr, ExcludedAttributes> {
    disabled?: boolean | undefined; // required but has default value
    invalid?: boolean | undefined;
    invalidText?: React.ReactNode | undefined;
    isOpen?: boolean | undefined;
    light?: boolean | undefined;
    size?: ListBoxSize | undefined;
    type?: ListBoxType | undefined; // required but has default value
    warn?: boolean | undefined;
    warnText?: React.ReactNode | undefined;
  }

  export interface ListBoxComponent
    extends ForwardRefReturn<HTMLDivElement, ListBoxProps> {
    readonly Field: ListBoxFieldComponent;
    readonly Menu: ListBoxMenuComponent;
    readonly MenuIcon: ListBoxMenuIconComponent;
    readonly MenuItem: ListBoxMenuItemComponent;
  }

  declare const ListBox: ListBoxComponent;

  export default ListBox;
}
```
I opened this PR because there was a problem, but even if this PR is not merged, there will be no problem in using it.

**Comment by jakebailey** [comment]
The tree shaking problem and TypeScript shouldn't have anything in common; a bundler will be repacking the output js files and won't be looking at `d.ts` files outside of maybe a typechecking pass. The only link between them is that there's no `d.ts` files for the `es` directory you're directly importing from the package.

I'm suggesting that you don't copy and paste all of the declarations, which will cause maintenance problems in the future because things are declared twice (so anyone updating this stub has to remember to repeat it). There are other packages which solve this dual ES/CJS distribution problem by re-exporting one declaration file from the other, which will turn out better, and not break your tree shaking case.

**Review by sandersn — CHANGES_REQUESTED** [review]
requesting changes based on @jakebailey 's comments.
